# HG changeset patch
# User Carlo de Falco <kingcrimson@tiscali.it>
# Date 1332277122 -3600
#      Tue Mar 20 21:58:42 2012 +0100
# Node ID ceb77e004f9ce6e4412a28d6c29b91da16d63540
# Parent  05b59be209ed5f2076040c53416d62e90a137e52
Fix editing file that shadows existing function

* script/miscellaneous/edit.m: fix pasting code of existing
 function into new file.

diff --git a/scripts/miscellaneous/edit.m b/scripts/miscellaneous/edit.m
--- a/scripts/miscellaneous/edit.m
+++ b/scripts/miscellaneous/edit.m
@@ -416,33 +416,33 @@ SUCH DAMAGE.\
     case {"cc", "C", "cpp"}
       if (isempty (head))
         comment = cstrcat ("/*\n", tail, "\n\n*/\n\n");
       else
         comment = cstrcat ("/*\n", head, "\n\n", tail, "\n\n*/\n\n");
       endif
       ## If we are shadowing an m-file, paste the code for the m-file.
       if (any (exists == [2, 103]))
-        code = cstrcat ("\\ ", strrep (type (name), "\n", "\n// "));
+        code = cstrcat ("\\ ", strrep (type (name){1}, "\n", "\n// "));
       else
         code = " ";
       endif
       body = cstrcat ("#include <octave/oct.h>\n\n",
                      "DEFUN_DLD(", name, ",args,nargout,\"\\\n",
                      name, "\\n\\\n\")\n{\n",
                      "  octave_value_list retval;\n",
                      "  int nargin = args.length();\n\n",
                      code, "\n  return retval;\n}\n");
 
       text = cstrcat (comment, body);
     case "m"
       ## If we are editing a function defined on the fly, paste the
       ## code.
       if (any (exists == [2, 103]))
-        body = type (name);
+        body = type (name){1};
       else
         body = cstrcat ("function [ ret ] = ", name, " ()\n\nendfunction\n");
       endif
       if (isempty (head))
         comment = cstrcat ("## ", name, "\n\n",
                           "## ", strrep (tail, "\n", "\n## "), "\n\n");
       else
         comment = cstrcat ("## ", strrep(head,"\n","\n## "), "\n\n", ...
